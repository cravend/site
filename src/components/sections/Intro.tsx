import { useTranslations } from "next-intl";

import styles from "../../styles/modules/sections/Intro.module.scss";
import LanguageToggle from "../LanguageToggle";

const Intro = () => {
  const t = useTranslations("sections.Intro");

  return (
    <>
      <LanguageToggle />
      <p>
        <span className={styles.greeting}>
          {t.rich("greeting", {
            Color: (children) => (
              <span className={styles.gradient}>{children}</span>
            ),
          })}
        </span>
        {t.rich("intro")}
      </p>
    </>
  );
};

export default Intro;
