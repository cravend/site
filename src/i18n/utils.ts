import { locales, defaultLocale, RTL_LOCALES } from "./config";

import type { Locale } from "./config";
import type { NextRouter } from "next/router";

/**
 * @description This function is used to check if a string is a valid locale.
 *
 * @param localeToTest The string to check.
 * @returns True if the string is a valid locale, false otherwise.
 */
export const isLocale = (localeToTest?: string): localeToTest is Locale =>
  locales.some((locale) => locale === localeToTest);

/**
 * @description This function is used to get the current locale.
 *
 * If a custom locale is set by \@next/router, it will be returned.
 * If no custom locale is set, the router's default locale will be returned.
 * If the router's default locale isn't set, the default locale will be returned.
 *
 * @param router The NextRouter instance.
 * @returns The current locale.
 */
export const getLocale = (router: NextRouter | null) => {
  if (!router) return defaultLocale;
  if (isLocale(router.locale)) return router.locale;
  if (isLocale(router.defaultLocale)) return router.defaultLocale;
  return defaultLocale;
};

/**
 * @description This function is used to toggle the current locale.
 *
 * @param locale The current locale to be toggled.
 * @returns The new locale.
 */
export const toggleLocale = (locale: Locale) =>
  locale === locales[0] ? locales[1] : locales[0];

/**
 * @description This function is used to get the text direction of the current locale.
 *
 * @param locale The locale to be checked
 * @returns True if the locale is read right-to-left, false otherwise
 */
export const isRtl = (locale: Locale) => RTL_LOCALES.includes(locale);
