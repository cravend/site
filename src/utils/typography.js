import Typography from 'typography'

const typography = new Typography({
  baseLineHeight: 1.666,
  headerFontFamily: ['Poppins', 'Helvetica', 'Arial', 'sans-serif'],
  bodyFontFamily: ['Poppins', 'Helvetica', 'Arial', 'sans-serif'],
  googleFonts: [
    {
      name: 'Poppins',
      styles: ['300', '600', '700']
    }
  ]
})

export default typography
