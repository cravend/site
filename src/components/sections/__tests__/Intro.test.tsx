import { render } from "../../../tests/utils";
import Intro from "../Intro";

describe("<Intro />", () => {
  it("renders", () => {
    expect.assertions(1);
    const { container } = render(<Intro />);
    expect(container).toMatchSnapshot();
  });
});
