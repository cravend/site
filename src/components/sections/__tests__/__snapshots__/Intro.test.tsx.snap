// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Intro /> renders 1`] = `
<div>
  <div
    class="container"
  >
    <a
      class="primary"
      href="/"
    >
      Lire en français
    </a>
  </div>
  <p>
    <span
      class="greeting"
    >
      Hello! My name is 
      <span
        class="gradient"
      >
        Dalton Craven
      </span>
      . 
    </span>
    It’s hard to introduce myself over the internet, but here’s a little bit about me:
  </p>
</div>
`;
